waku/v2/waku_archive/driver/postgres_driver/postgres_driver.nim

```diff
@@ -0,0 +1,288 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/db_postgres,
+  std/strformat,
+  std/nre,
+  std/options,
+  std/strutils,
+  stew/[results,byteutils],
+  chronos
+
+import
+  ../../../waku_core,
+  ../../common,
+  ../../driver
+
+export postgres_driver
+
+type PostgresDriver* = ref object of ArchiveDriver
+  connection: DbConn
+  preparedInsert: SqlPrepared
+
+proc dropTableQuery(): string =
+  "DROP TABLE messages"
+
+proc createTableQuery(): string =
+  "CREATE TABLE IF NOT EXISTS messages (" &
+  " pubsubTopic VARCHAR NOT NULL," &
+  " contentTopic VARCHAR NOT NULL," &
+  " payload VARCHAR," &
+  " version INTEGER NOT NULL," &
+  " timestamp BIGINT NOT NULL," &
+  " id VARCHAR NOT NULL," &
+  " storedAt BIGINT NOT NULL," &
+  " CONSTRAINT messageIndex PRIMARY KEY (storedAt, id, pubsubTopic)" &
+  ");"
+
+proc insertRow(): string =
+ """INSERT INTO messages (id, storedAt, contentTopic, payload, pubsubTopic,
```

Comment:
will this insert work if we try to insert a duplicated message? I mean, insert a message twice. Perhaps we need ON CONFLICT?

mind adding a testcase if its the case?

Reply:
Thanks for the comment @alrevuelta 
I think it should be added correctly as the primary key would be different, given different timestamps (nanosecconds precision)
The expected behavior is that the duplicate messages are properly added right ?

Reply:
In case of completely equal messages, the `put` call will return an error. I will add a little test with the next:
```code
putRes = await driver.put(DefaultPubsubTopic,
                              msg2, computeDigest(msg2), msg2.timestamp)
    ## Then
    require:
      not putRes.isOk()
```